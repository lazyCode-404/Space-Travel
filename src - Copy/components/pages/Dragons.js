import React from 'react';

const Dragons = () => {
  const fetchDragons = () => {
    fetch('https://api.spacexdata.com/v3/dragons')
      .then((res) => res.json())
      .then((json) => console.log(json));
  };
  fetchDragons();
  return (
    <div>Dragons</div>
  );
};

export default Dragons;
