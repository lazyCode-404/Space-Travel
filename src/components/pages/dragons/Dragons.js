import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchDragons, reserve, cancelReservation } from '../redux/dragons/dragons.';
import styles from './styles/dragons.module.css';

export default function Dragons() {
  const dragons = useSelector((state) => state.dragons.dragons);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!dragons) {
      dispatch(fetchDragons());
    }
  }, []);

  const handleReserve = (id) => {
    dispatch(reserve(id));
  };

  const cancelReserve = (id) => {
    dispatch(cancelReservation(id));
  };

  return (
    <section className={styles.main}>
      {dragons?.map((drag) => (
        <div key={drag.id} className={styles.dragonList}>
          <div>
            <img src={drag.images} className={styles.images} alt="" />
          </div>
          <div>
            <h1>{drag.name}</h1>
            <p>{drag.type}</p>
            <div className={styles.cardItems}>
              <h3>{drag.names}</h3>
              <p>
                <span className={drag.reserved ? styles.badge : styles.noBadge}>
                  Reserved
                </span>
                {drag.description}
              </p>
              {drag.reserved ? (
                <button
                  id={drag.id}
                  className={styles.cancelDragon}
                  type="button"
                  onClick={(e) => {
                    cancelReserve(e.target.id);
                  }}
                >
                  Cancel Reservation
                </button>
              ) : (
                <button
                  id={drag.id}
                  className={styles.button}
                  type="button"
                  onClick={(e) => {
                    handleReserve(e.target.id);
                  }}
                >
                  Reserve Dragon
                </button>
              )}
            </div>
          </div>
        </div>
      ))}
    </section>
  );
}
