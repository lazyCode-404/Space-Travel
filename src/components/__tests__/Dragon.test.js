import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Dragons from '../pages/dragons/Dragons';
import store from '../../redux/configureStore';

it('Missions component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Dragons />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
