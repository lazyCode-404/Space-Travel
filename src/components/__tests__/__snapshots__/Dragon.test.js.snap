// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component 1`] = `
<section
  className="main"
/>
`;
